tools/Dashboard/DevHome.Dashboard/ViewModels/WidgetViewModel.cs

```diff
@@ -312,42 +310,36 @@ private Grid GetErrorCard(string error, string subError = null)
         return grid;
     }
 
-    private JObject WrapJsonString(string jsonString)
+    private Newtonsoft.Json.Linq.JObject WrapJsonString(string jsonString)
     {
-        return new JObject { ["data"] = jsonString };
+        return new Newtonsoft.Json.Linq.JObject { ["data"] = jsonString };
     }
 
-    private string MergeJsonData(string jsonStringA, string jsonStringB)
+    private string MergeJsonData(Windows.Data.Json.JsonValue actionValue, Windows.Data.Json.JsonObject inputsObject)
     {
-        if (string.IsNullOrEmpty(jsonStringA))
-        {
-            return jsonStringB;
-        }
+        Newtonsoft.Json.Linq.JObject objA = [];
+        Newtonsoft.Json.Linq.JObject objB = [];
 
-        if (string.IsNullOrEmpty(jsonStringB))
+        if (actionValue != null)
         {
-            return jsonStringA;
-        }
-
-        JObject objA;
-        JObject objB;
-
-        try
-        {
-            objA = JObject.Parse(jsonStringA);
-        }
-        catch (JsonReaderException)
-        {
-            objA = WrapJsonString(jsonStringA);
+            var dataType = actionValue.ValueType;
+            if (dataType == Windows.Data.Json.JsonValueType.Object)
+            {
+                objA = Newtonsoft.Json.Linq.JObject.Parse(actionValue.Stringify());
+            }
+            else if (dataType == Windows.Data.Json.JsonValueType.String)
+            {
+                objA = WrapJsonString(actionValue.Stringify());
+            }
         }
 
-        try
+        if (inputsObject != null)
         {
-            objB = JObject.Parse(jsonStringB);
-        }
-        catch (JsonReaderException)
-        {
-            objB = WrapJsonString(jsonStringB);
+            var inputType = inputsObject.ValueType;
+            if (inputType != Windows.Data.Json.JsonValueType.Null)

```

Comment:
Is line 339, 340, 341, and 342 tabbed with "\t"?

Reply:
No, I think the color in GitHub comes from it being indented one more time.